### PR TITLE
history: add `!n` and `!?string?` commands

### PR DESCRIPTION
**Pull request by jopefd**
- [ ] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [ ] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/history.md [hunk 1]
```diff
@@ -7,6 +7,14 @@
 
 `history`
 
+- Run the n-th command of the history list:
```
Comment (bl-ue) [inline pages/common/history.md:10]:
```suggestion
- Run the Nth command of the history list:
```

**Review by bl-ue — APPROVED** [review]
Cool! Looks good to me, except I'd put `n-th` differently...

File: pages/common/history.md (hunk 1 above)
Comment (jopefd) [reply pages/common/history.md:10]: Ok! I gonna fix it :)

**Review by bl-ue — APPROVED** [review]
Indeed, these work for me.  Interesting!

**Comment by jopefd** [comment]
> Indeed, these work for me. Interesting!

Good! =D

File: pages/common/history.md (hunk 1 above)
Comment (marchersimon) [reply pages/common/history.md]: Actually, since all our clients support UTF-8, why not "Nᵗʰ"?

**Review by marchersimon — APPROVED** [review]
Thanks @jopefd

**Comment by CleanMachine1** [comment]
Just a heads up, open a branch on your fork with the relevant name, instead of pushing to main branch, just in case you ever become a collaborator and good practise to not push straight to main

**Review by CleanMachine1 — APPROVED** [review]

**Comment by jopefd** [comment]
> Just a heads up, open a branch on your fork with the relevant name, instead of pushing to main branch, just in case you ever become a collaborator and good practise to not push straight to main

Ah, understood! Next time I'll do this. Thank you!

File: pages/common/history.md (hunk 1 above)
Comment (jopefd) [reply pages/common/history.md]: For me Nᵗʰ is ok too. Should I change to it? @bl-ue @CleanMachine1

Comment (marchersimon) [reply pages/common/history.md]: I'm not sure anymore. In monospace, which terminals use, it looks like this: `Nᵗʰ`

Comment (CleanMachine1) [reply pages/common/history.md]: hmmm I like the smaller versions however as mentioned, special characters like that often look awful under most fonts.

**Review by navarroaxel — CHANGES_REQUESTED** [review]
I think `histexpand` is the place for these new examples. @sbrl?

**Comment by marchersimon** [comment]
We should've mentioned the `histexpand` page in `history.md`. `histexpand` already has 8 examples, but there we have `!-n`, which runs the command n lines back (which is in my opinion more useful than `!n`) and 

> \- Run the most recent command starting with string:
    !string

which we could replace with `!?string?`, since it's more universal.

**Comment by jopefd** [comment]
> We should've mentioned the `histexpand` page in `history.md`. `histexpand` already has 8 examples, but there we have `!-n`, which runs the command n lines back (which is in my opinion more useful than `!n`) and
> 
> > - Run the most recent command starting with string:
> > !string
> 
> which we could replace with `!?string?`, since it's more universal.

1. I think it's better to offer options, and let the user choose which is better for its case. In my case I don't use `!-n`, so I'd choose `!n`.
2. Yes, `!?string?` covers `!string`, so it seems me better.

_Note: `!?string?` vs. `!string` and `!-n` vs. `!n` are two distinct cases to me. While `!?string?` covers `!string`, neither `!-n` nor `!n` covers the other. `!n` works for fixed command ids and `!-n` works for relative command ids. That's why I prefer that the user can choose._

**Comment by marchersimon** [comment]
We have to decide for either `!n` or `!-n`, since pages can't have more than 8 examples. Personally I don't use either of them, so I can't judge which would be more helpful. Regarding `!?string?` vs `!string` I'm also not really sure. The first covers the second, but I can imagine that in most cases you want to get the command which starts with a specific string (`!string`).

**Comment by jopefd** [comment]
Alright...

`!?string?` still seems me to be the best option, because, for example, it allows me to search `!?sudo ln -s?`. It wouldn't be uselful search `!sudo`; it'll give me any command runned as sudo...

About `!-n!` to me it's not useful. I don't have any need to run this, and I think _"why does someone need to calculate some command id for execute it? It's just execute `history`, see the id, and execute `!id`"... no calculations_. But I don't know if there is some relevant use for `!-n`...

What do you think? @navarroaxel @CleanMachine1 @bl-ue

**Comment by CleanMachine1** [comment]
I don't really know. However I am watching this PR.

_just  leaving my little notice that im not continuing this on my end_

**Comment by sbrl** [comment]
@navarroaxel: Yeah, I would probably say that `histexpand` is the correct place for these examples. `history` should definitely have a `See also` though pointing to `histexpand`.

**Comment by navarroaxel** [comment]
@jopefd don't close this PR. Just revert `history` and move the changes to `histexpand`.

**Comment by marchersimon** [comment]
I'll close this now, since all changes proposed here are made in #6051

**Pull request closed without merging**